OptimizelySDKCore/OptimizelySDKCore/OPTLYProjectConfig.m

```diff
@@ -108,11 +110,20 @@ - (instancetype)initWithBuilder:(OPTLYProjectConfigBuilder *)builder {
         return nil;
     }
     
+    supportedDatafileVersions = @[@"2", @"3", @"4"];
+    
     // check datafile is valid
     @try {
         NSError *datafileError;
         OPTLYProjectConfig *projectConfig = [[OPTLYProjectConfig alloc] initWithData:builder.datafile error:&datafileError];
         
+        if (!datafileError && ![supportedDatafileVersions containsObject:projectConfig.version]) {
```

Comment:
Is this `!datafileError` check needed? Seems like we only need the second one.

Reply:
I think it would be necessary in case so other parse error occurs.